### PR TITLE
feat(prover-sever): enhance error checking

### PR DESCRIPTION
**Pull request by lightscale-luke**
* update constants related to generate proof. (e.g., MAX_TXS, MAX_CALLDATA, etc)
* check if tx count from trace is smaller than MAX_TXS before starting proof creation.
* check if chain id in trace equals to prover's chain id.
* if requested with an incorrect prove type, prover returns error before initiating a trace object.
* detailed an error message in response.

### CODE REVIEW

File: bin/src/prove.rs [hunk 1]
```diff
@@ -78,6 +78,16 @@ fn main() {
     // Generating proofs for each trace
     let mut outer_timer = Measurer::new();
     for (trace_name, trace) in traces {
+        let tx_count = trace.transactions.len();
+        if tx_count > MAX_TXS {
+            panic!(
+                "{}",
+                format!(
+                    "Too many transactions. MAX_TXS:{}, given transactions:{}",
```
Comment (fakedev9999) [inline bin/src/prove.rs]:
```suggestion
                    "too many transactions. MAX_TXS:{}, given transactions:{}",
```
let's use lowercase for consistency

File: prover-server/src/server_main.rs [hunk 2]
```diff
@@ -58,6 +59,15 @@ impl Rpc for RpcImpl {
                 return Err(err);
             }
         };
+        let tx_count = block_trace.transactions.len();
+        if tx_count > MAX_TXS {
+            kroma_err(format!(
+                "Too many transactions. MAX_TXS:{}, given transactions:{}",
```
Comment (fakedev9999) [inline prover-server/src/server_main.rs]:
```suggestion
                "too many transactions. MAX_TXS:{}, given transactions:{}",
```

**Review by fakedev9999 — APPROVED** [review]
LGTM

**Review by fakedev9999 — APPROVED** [review]
LGTM

**Comment by fakedev9999** [comment]
@dongchangYoo For the commit message "[feat(prover-server): increase request size to 32Mib](https://github.com/kroma-network/kroma-prover/pull/20/commits/4a934d15108611c2725cea62696003b33302d172)" can you fix it to Mb?

**Comment by chokobole** [comment]
Please change commit title `feat: apply new values to params` to `feat(zkevm): apply new values to params`

**Comment by chokobole** [comment]
Could you add the reason why you increase request size to 32 Mib in commit body?

**Comment by dongchangYoo** [comment]
1 MiB stands for 1,000,000 bytes. However, it would be more accurate to express it as MiB instead of Mib.

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by fakedev9999 — APPROVED** [review]
LGTM

**Review by chokobole — APPROVED** [review]
LGTM

**Pull request merged**